Assets/Scripts/States/ChasingState.cs

```diff
@@ -5,33 +5,40 @@ public class ChasingState : TankState
 {
     float speed = 10;
     bool Enterstate = false;
-    Transform target = GameObject.FindGameObjectWithTag("Player").transform;
+    Transform target;
+    float timelapse = 0;
     private void Update()
     {
         if (Enterstate)
         {
-            enemyBehaviour.transform.position = Vector3.MoveTowards(enemyBehaviour.transform.position, target.position, 10 * Time.deltaTime);                       
-        } 
-        if(Vector3.Distance(enemyBehaviour.transform.position,target.position)>2.5f)
-        {
-            enemyBehaviour.ChangeState(enemyBehaviour.patrollingState);
+            enemyBehaviour.transform.position = Vector3.MoveTowards(enemyBehaviour.transform.position, target.position, speed * Time.deltaTime);
+            if (timelapse > 3)
+            {
+                Debug.LogError("Change To Patrolling State");
+                enemyBehaviour.ChangeState(enemyBehaviour.patrollingState);
+            }
+            else
+            {
+                timelapse += Time.deltaTime;
+            }
         }
+        
     }
     public override void Awake()
     {
         base.Awake();
+        target = GameObject.FindGameObjectWithTag("Player").transform;

```

Comment:
why using tag to find the player?

Reply:
only when a enemy comes in contact with the player (through collision) you should get access to the player then and only then